packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -195,11 +202,66 @@ export class DataStores implements IDisposable {
 		return pendingAliasPromise ?? "Success";
 	}
 
+	/** For sampling. Only log once per container */
+	private shouldSendAttachLog = true;
+
+	/** @returns true if it found/processed GC Data, false otherwise */
+	private extractAndProcessAttachGCData(id: string, entries?: ITreeEntry[]): boolean {
+		const gcDataEntry = entries?.find((e) => e.path === ".gcdata");
+
+		// Old attach messages won't have GC Data (And REALLY old ones won't even have a snapshot ITree!)
+		if (gcDataEntry === undefined) {
+			return false;
+		}
+
+		assert(
+			gcDataEntry.type === TreeEntry.Blob && gcDataEntry.value.encoding === "utf-8",
+			"GC data should be a utf-8-encoded blob",
+		);
+
+		const gcData = JSON.parse(gcDataEntry.value.contents) as IGarbageCollectionData;
+		for (const [nodeId, outboundRoutes] of Object.entries(gcData.gcNodes)) {
+			const fromHandle = {
+				absolutePath: `/${id}${nodeId === "/" ? "" : nodeId}`,
+			};
+			outboundRoutes.forEach((route) => {
+				const toHandle = {
+					absolutePath: `${route}`,
+				};
+
+				this.runtime.addedGCOutboundReference(fromHandle, toHandle);
+			});
+		}
+		return true;
+	}
+
 	public processAttachMessage(message: ISequencedDocumentMessage, local: boolean) {
 		const attachMessage = message.contents as InboundAttachMessage;
 
 		this.dataStoresSinceLastGC.push(attachMessage.id);
 
+		// We need to process the GC Data for both local and remote attach messages
+		const foundGCData = this.extractAndProcessAttachGCData(
+			attachMessage.id,
+			attachMessage.snapshot?.entries,
+		);
+
+		// Only log once per container to avoid noise/cost.
+		// Allows longitudinal tracking of various state (e.g. foundGCData), and some sampled details
+		if (this.shouldSendAttachLog) {
+			this.shouldSendAttachLog = false;
+			this.mc.logger.sendTelemetryEvent({
```

Comment:
Should you log this only if `foundGCData` is true? So that we dont end up not getting the data for that case.

Reply:
I don't care to see every single hit here. The hypothesis is that over time as Loop deploys the bump with this change, we'll see more and more and more with foundGCData true: I don't expect to do much more analysis beyond that, and don't want to log on every attach x N collaborators

Reply:
Circling back - I _do_ want some logs with `foundGCData` false.  They will start out as the majority but then get more and more rare.